### ISSUE TITLE
Apache examples use %b

### ISSUE THREAD

**Issue by giganteous** [google/mtail#147]
Hi!

Thanks for maintaining this as open source!

I discovered a minor issue with the **apache (combined and common) examples** where these trie
to parse %b. This fails on empty response sizes (which happen when e.g. 304's are returned). The [apache documentation](http://httpd.apache.org/docs/2.4/logs.html#accesslog) suggests this:

```
2326 (%b)**
The last part indicates the size of the object returned to the client, not including the response headers. If no content was returned to the client, this value will be "-". To log "0" for no content, use %B instead.
```

I propose to try parse %b and fix the regexp to allow for a single '-':

```
...
/((?P<response_size>\d+)|-) / + # %b
...
  $response_size > 0 {
      apache_http_bytes_total[$request_method][$http_version][$request_status] += $response_size
  }
...
```
I can make a PR fixing this if that helps?

**Comment by jaqx0r** [comment]
Thanks for the report, a PR would be perfect! Can you add a change to the
ex_test.go golden files to test it as well?

**Comment by giganteous** [comment]
I will. Can you elaborate on how to reliably provide a "Datum" in the .golden file, or a pointer to where I can learn this a little? :)

I'm stuck on this test output (the rest I could manage to write down by hand).

Thanks!

```
--- FAIL: TestExamplePrograms (0.01s)
    --- FAIL: TestExamplePrograms/examples/apache_combined.mtail_on_testdata/apache-combined.log (0.01s)
    	ex_test.go:147: {*metrics.Store}.Metrics["apache_http_bytes_total"][0].LabelValues[0].Value.BaseDatum.Time:
    			-: 1521809199311215243
    			+: 1521809199310035186
    		{*metrics.Store}.Metrics["apache_http_requests_total"][0].LabelValues[0].Value.BaseDatum.Time:
    			-: 1521809199311239209
    			+: 1521809199310014777
    		{*metrics.Store}.Metrics["apache_http_requests_total"][0].LabelValues[1].Value.BaseDatum.Time:
    			-: 1521809199311259369
    			+: 1521809199310080225
```

**Comment by jaqx0r** [comment]
Sorry for the delay.

In the other test examples (e.g examples/rsyncd.mtail) you can see I hard
code a timestamp in the top of the example program (with a comment
explaining why.)  This keeps the test runs stable.

**Comment by jaqx0r** [comment]
Did you make any progress on this?

**Comment by giganteous** [comment]
I did, but didn't realize that it was still waiting on my side. Having a look!

**Comment by giganteous** [comment]
This was fixed in #150

**Comment by jaqx0r** [comment]
Haha, yes. I was quite confused!
